templates/hooks/db_instance/sdk_update_pre_build_request.go.tpl

```diff
@@ -14,7 +14,7 @@
 		ackcondition.SetSynced(desired, corev1.ConditionTrue, nil, nil)
 		return desired, nil
 	}
-	if !instanceAvailable(latest) {
+	if !instanceAvailable(latest) && !strings.Contains(*latest.ko.Status.DBInstanceStatus, "storage-full") && !delta.DifferentAt("Spec.AllocatedStorage") {
```

Comment:
nit: Can we create a helper function called `needStorageUpdate` and we do something like:
```
if !instanceAvailable(latest) && if !needStorageUpdate(latest) { ... }
```